api/registry/api.py

```diff
@@ -98,6 +101,28 @@ class ErrorMessageResponse(Schema):
     detail: str
 
 
+def check_rate_limit(request):
+    """
+    Check the rate limit for the API.
+    This is based on the original ratelimit decorator from django_ratelimit
+    """
+    old_limited = getattr(request, "limited", False)
+    rate = request.api_key.rate_limit
+    ratelimited = is_ratelimited(
+        request=request,
+        group="registry",
+        fn=None,
+        key="header:x-api-key",
+        rate=rate,
+        method=ALL,
+        increment=True,
+    )
+    request.limited = ratelimited or old_limited
+    if ratelimited:
+        cls = getattr(settings, "RATELIMIT_EXCEPTION_CLASS", Ratelimited)
+        raise (import_string(cls) if isinstance(cls, str) else cls)()
```

Comment:
can we convert this to a decorator?

Reply:
That was actually my first try, but I did not get it to work.
The problem is that we need this to be run after the auth is applied, but before the event handler .... 
And I was not able to get this behaviour, because auth is handled in another decorator that also calls the handler function.

Reply:
valid point... we can keep it this way